pkg/controller/shard_handler.go

```diff
@@ -0,0 +1,79 @@
+package controller
+
+import (
+	"context"
+	typeV1 "github.com/istio-ecosystem/admiral-api/pkg/apis/admiral/v1"
+	"github.com/istio-ecosystem/admiral-sharding-manager/pkg/model"
+	log "github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ShardInteface interface {
+	CreateOrUpdateShard(ctx context.Context, shard *typeV1.Shard) (*typeV1.Shard, error)
```

Comment:
We could just name the functions as: `Create`, `Update`, `Delete`, given they are already part of the ShardInterface.

Reply:
sure will update these